graphql-codegen-haskell/src/render/templates/api.mustache

```diff
@@ -57,10 +62,6 @@ instance GraphQLQuery {{queryName}} where
     {{queryText}}
   |]
 
-  getArgs {{queryName}}{{argsRecordSyntax}} = object
-    {{#overArgs}}
-    [ "{{arg}}" .= _{{arg}}
-    {{/overArgs}}
-    ]
+  getArgs {{queryName}}{{argsRecordSyntax}} = toJSON {{queryName}}{{argsRecordSyntax}}
```

Comment:
Leverage the automatically derived `ToJSON` instance to generate the JSON for `getArgs`.